### ISSUE TITLE
[@types/react-filepond] onprocessfiles Callback property missing

### ISSUE THREAD

**Issue by SumAznGuy** [DefinitelyTyped/DefinitelyTyped#43601]
- [x] I tried using the `@types/xxxx` package and had problems.
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @zposten

According to the FilePond documentation, there's a `onprocessfiles` prop which is missing from the FilePondCallbackProps type definition.
https://pqina.nl/filepond/docs/patterns/api/filepond-instance/

Apparently there been a definition for this in the base filepond library:
https://github.com/pqina/filepond/blob/2f547d1b475e7c0c658917e7d6f7714005169067/types/index.d.ts#L1113

**Comment by guicara** [comment]
Hello,

Any news on this issue?

In the meantime, I have extended the type definitions:

```
// tsconfig.json
{
  "compilerOptions": {
    ...
    "typeRoots": [
      "src/types",
      "node_modules/@types"
    ]
  },
}
```

```
// src/types/ReactFilepond.d.ts

// Extended type definitions for react-filepond 5.0
// Project: https://github.com/pqina/react-filepond, https://pqina.nl/filepond
// Definitions by: Zach Posten <https://github.com/zposten>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
// TypeScript Version: 2.8

import { FilePondProps } from 'react-filepond';

declare module 'react-filepond' {
    interface FilePondCallbackProps {
        /** Called when all files in the list have been processed. */
        onprocessfiles?: () => void;
    }
}
```

Another "solution" _(but definitely not a good one)_ is to use `onupdatefile` instead of the missing `onupdatefiles`. Something like that:

```
const handleOnProcessFile = (error: any, file: File) => {
    let countCompleted = 0;
    files.forEach((f) => {
        if (f && f.status === FileStatus.PROCESSING_COMPLETE) {
            countCompleted += 1;
        }
    });

    // Check if all the files have been uploaded without errors
    if (countCompleted === files.length && onFilesUploaded) {
        // call your callback function here...
    }
};
```
Hope it helps.

**Comment by zposten** [comment]
Fixed by #47008

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
